libblkid/src/probe.c

```diff
@@ -996,6 +1021,7 @@ int blkid_probe_set_device(blkid_probe pr, int fd,
 	pr->fd = fd;
 	pr->off = (uint64_t) off;
 	pr->size = 0;
+	pr->io_size = 0;
```

Comment:
It would be better to default to 512.

Reply:
DEFAULT_SECTOR_SIZE would probably be better :-)